Add PR release note label checker bot

jackzhxng opened the pull request:
### Summary
Add gh actions workflow for checking that there is a `Release notes:` or `topic: not user facing` label is added to each PR.

### Test plan
Testing with this PR

guangy10 commented on .github/workflows/check-labels.yml:
```diff
@@ -0,0 +1,52 @@
+  # to test changes to the workflow itself one needs to create
+  # a PR that targets a gh/**/base branch.
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches: [gh/**/base]
```
I think you may want to test is on regular PR (not created via ghstack) as well by removing this temporarily. We should see it run and catch.

dvorjackz replied:
Wouldn't we want to keep `main` here as well since? Otherwise it will only run when using `ghstack`

huydhn replied:
Nope, we need to remove main here because it's a security risk.  This workflow is granted write access, so it needs to get the `check_labels.py` script from main as part of `pull_request_target` trigger.  Basically, it will grant write access to random GitHub users (privileges escalation).  You see the ghstack part here because folks who are using ghstack already have write access to the repo.

huydhn replied:
And don't worry about the check not running in your PR, you can land this and it will start running on all PRs after

guangy10 commented on .github/workflows/check-labels.yml:
```diff
@@ -0,0 +1,52 @@
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches: [main]
+
+  # To check labels on ghstack PRs.
+  # Note: as pull_request doesn't trigger on PRs targeting main,
+  # to test changes to the workflow itself one needs to create
+  # a PR that targets a gh/**/base branch.
+  pull_request:
```
Can you create this PR from codev by exporting it using ghstack so you can test on a gh/ branch?

dvorjackz replied:
https://github.com/pytorch/executorch/pull/6930 for ghstack

Commit:
Add release note label checker bot

ghstack-source-id: cb91383998bced56a74c58e2e5afae092152b694
ghstack-comment-id: 2483597745
Pull Request resolved: https://github.com/pytorch/executorch/pull/6927

Commit:
Fix formatting (?)

ghstack-source-id: 58a1e6e90c95ffc59fd5fc2ce1881c24dc032ea9
ghstack-comment-id: 2483597853
Pull Request resolved: https://github.com/pytorch/executorch/pull/6928

Commit:
Try to activate test on non ghstack pr

huydhn committed:
Fix indentation

Commit:
Resolve mising trymerge deps

huydhn commented on .github/workflows/check-labels.yml:
```diff
@@ -0,0 +1,54 @@
+    if: github.repository_owner == 'pytorch'
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
```
Let me double check the behavior of `actions/checkout` a bit to make sure that this is secure

huydhn commented on .github/workflows/check-labels.yml:
```diff
@@ -0,0 +1,54 @@
+  # a PR that targets a gh/**/base branch.
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches: [gh/**/base, main]
```
You cannot use main here as it poses a security risk. See https://fb.workplace.com/groups/pytorch.dev/permalink/1658227814755692.
```suggestion
    branches: [gh/**/base]
```

huydhn committed:
Update .github/workflows/check-labels.yml

Cannot use main as part of pull_request target here.  See https://fb.workplace.com/groups/pytorch.dev/permalink/1658227814755692

kirklandsign commented:
> To add a label, you can comment to pytorchbot, for example
@pytorchbot label "topic: not user facing"


Does it work?

dvorjackz commented:
@kirklandsign were some issues with the comment deletion - please see https://github.com/pytorch/executorch/pull/6953
